sourcetracker/_cli/gibbs.py

```diff
@@ -134,96 +131,69 @@ def gibbs(table_fp, mapping_fp, output_dir, loo, jobs, alpha1, alpha2, beta,
 
-    # Build function that require only a single parameter -- sample -- to
-    # enable parallel processing if requested.
-    if loo:
-        f = partial(_cli_loo_runner, source_category=source_category_column,
-                    alpha1=alpha1, alpha2=alpha2, beta=beta,
-                    restarts=restarts, draws_per_restart=draws_per_restart,
-                    burnin=burnin, delay=delay,
-                    sample_metadata=sample_metadata,
-                    sources_data=sources_data, sources_envs=sources_envs,
-                    biom_table=biom_table, output_dir=output_dir)
-        sample_iter = source_samples
-    else:
-        f = partial(_cli_sink_source_prediction_runner, alpha1=alpha1,
-                    alpha2=alpha2, beta=beta, restarts=restarts,
-                    draws_per_restart=draws_per_restart, burnin=burnin,
-                    delay=delay, sources_data=sources_data,
-                    biom_table=biom_table, output_dir=output_dir)
-        sample_iter = sink_samples
+    # Prepare sink data and rarify if requested.
+    sinks = feature_table.loc[sink_samples, :]
+    if sink_rarefaction_depth > 0:
+        sinks = subsample_dataframe(sinks, sink_rarefaction_depth)
 
+    # If we've been asked to do multiple jobs, we need to spin up a cluster.
     if jobs > 1:
         # Launch the ipcluster and wait for it to come up.
         subprocess.Popen('ipcluster start -n %s --quiet' % jobs, shell=True)
         time.sleep(cluster_start_delay)
         c = Client()
-        c[:].map(f, sample_iter, block=True)
-        # Shut the cluster down. Answer taken from SO:
-        # http://stackoverflow.com/questions/30930157/stopping-ipcluster-engines-ipython-parallel
-        c.shutdown(hub=True)
     else:
-        for sample in sample_iter:
-            f(sample)
+        c = None
 
-    # Format results for output.
-    samples = []
-    samples_data = []
-    for sample_fp in glob.glob(os.path.join(output_dir, '*')):
-        samples.append(sample_fp.strip().split('/')[-1].split('.txt')[0])
-        samples_data.append(np.loadtxt(sample_fp, delimiter='\t'))
-    mp, mps = _cli_collate_results(samples, samples_data, sources_envs)
+    # Run the computations.
+    if not loo:
+        mpm, mps, fas = _gibbs(csources, sinks, alpha1, alpha2, beta, restarts,
+                               draws_per_restart, burnin, delay, cluster=c)
+    else:
+        mpm, mps, fas = _gibbs_loo(csources, alpha1, alpha2, beta, restarts,
+                                   draws_per_restart, burnin, delay, cluster=c)
+    # If we started a cluster, shut it down.
+    if jobs > 1:
+        c.shutdown(hub=True)
 
-    o = open(os.path.join(output_dir, 'mixing_proportions.txt'), 'w')
-    o.writelines(mp)
-    o.close()
-    o = open(os.path.join(output_dir, 'mixing_proportions_stds.txt'), 'w')
-    o.writelines(mps)
-    o.close()
+    # Write results.
+    mpm.to_csv(os.path.join(output_dir, 'mixing_proportions.txt'), sep='\t')
```

Comment:
I think I looked quickly and didn't find anything. Let me know if you want me to look into this more - or maybe searching StackOverflow would be a good way to proceed here.